In [29]:
library(readr)
library(readxl)
library(olsrr)
library(MASS)
library(normtest)
library(nortest)
library(psych)
library(fitdistrplus)
library(lattice)
library(survival)
library(Formula)
library(ggplot2)
library("Hmisc")
library(dplyr)

In [30]:
casen <- read_excel('CASEN_DB.xlsx')
head(casen)
n = length(casen$X1)

X1     X2     X3     X4     X5      X6     V1 V2 Z1 E1 E2 R O
1 119410 260000 854170  83340  778376 100000 1  0  1  1  0  1 1
2  79610 100000 975000 125000  585961 110000 0  1  1  1  0  0 0
3  39800 700000 662500 250000 2838385 120000 1  0  1  1  0  0 1
4  39800 700000 662500 250000 2838385 120000 1  0  1  1  0  0 1
5 119410 320000 500000  83330  789907 100000 1  0  1  1  0  0 1
6  79610 700000 733330  83330 1131115 120000 1  0  1  1  0  0 1

In [31]:
mod11 <- lm(X1 ~ X3 +X4+X6 +V1+V2+R+O-1, data=casen)
mod12 <- lm(log(X1) ~X2 +X3+X4+X6+V1+V2+Z1+E1+E2+R+O-1, data= casen)
mod14 <- lm(X1 ~ X3+X4+X6+V2+R+O-1, data=casen)

------------------
-------------------
### Recuento de los ajustes y su normalidad de residuales
---------------

In [32]:
summary(mod11)


Call:
lm(formula = X1 ~ X3 + X4 + X6 + V1 + V2 + R + O - 1, data = casen)

Residuals:
   Min     1Q Median     3Q    Max 
-94294 -25196   1139  22171 112228 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X3 3.065e-02  5.287e-03   5.797 2.95e-08 ***
X4 2.107e-01  3.945e-02   5.340 2.76e-07 ***
X6 9.808e-02  2.993e-02   3.277  0.00126 ** 
V1 1.228e+04  7.869e+03   1.561  0.12028    
V2 1.795e+04  6.561e+03   2.736  0.00685 ** 
R  1.042e+04  5.530e+03   1.885  0.06108 .  
O  2.026e+04  6.777e+03   2.989  0.00319 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 38430 on 181 degrees of freedom
Multiple R-squared:  0.8781,	Adjusted R-squared:  0.8734 
F-statistic: 186.3 on 7 and 181 DF,  p-value: < 2.2e-16


In [33]:
jb.norm.test(residuals(mod11))
lillie.test(residuals(mod11))


	Jarque-Bera test for normality

data:  residuals(mod11)
JB = 1.476, p-value = 0.4225



	Lilliefors (Kolmogorov-Smirnov) normality test

data:  residuals(mod11)
D = 0.047716, p-value = 0.371


In [34]:
summary(mod12)


Call:
lm(formula = log(X1) ~ X2 + X3 + X4 + X6 + V1 + V2 + Z1 + E1 + 
    E2 + R + O - 1, data = casen)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.9742 -0.9237  0.1483  1.3109  5.1271 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X2 3.782e-06  8.922e-07   4.239 3.61e-05 ***
X3 1.473e-06  2.599e-07   5.668 5.78e-08 ***
X4 7.475e-06  1.930e-06   3.872 0.000152 ***
X6 2.745e-06  1.492e-06   1.839 0.067545 .  
V1 1.973e+00  4.287e-01   4.603 7.91e-06 ***
V2 2.550e+00  3.350e-01   7.611 1.55e-12 ***
Z1 1.062e+00  3.177e-01   3.342 0.001015 ** 
E1 1.956e+00  3.475e-01   5.630 6.94e-08 ***
E2 2.118e+00  5.033e-01   4.208 4.08e-05 ***
R  8.210e-01  2.729e-01   3.009 0.003007 ** 
O  3.159e+00  3.805e-01   8.301 2.57e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.861 on 177 degrees of freedom
Multiple R-squared:  0.9749,	Adjusted R-squared:  0.9733 
F-statistic: 624.3 on 11 and 177 DF,  p-value: < 2.2e-16


In [35]:
jb.norm.test(residuals(mod12))
lillie.test(residuals(mod12))


	Jarque-Bera test for normality

data:  residuals(mod12)
JB = 2.1365, p-value = 0.2715



	Lilliefors (Kolmogorov-Smirnov) normality test

data:  residuals(mod12)
D = 0.054606, p-value = 0.1866


In [36]:
summary(mod14)


Call:
lm(formula = X1 ~ X3 + X4 + X6 + V2 + R + O - 1, data = casen)

Residuals:
    Min      1Q  Median      3Q     Max 
-102450  -24292    3237   23330  110070 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X3 3.051e-02  5.307e-03   5.749 3.73e-08 ***
X4 2.151e-01  3.950e-02   5.445 1.66e-07 ***
X6 1.067e-01  2.953e-02   3.613 0.000392 ***
V2 1.205e+04  5.384e+03   2.238 0.026454 *  
R  1.149e+04  5.509e+03   2.085 0.038468 *  
O  2.485e+04  6.129e+03   4.055 7.44e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 38580 on 182 degrees of freedom
Multiple R-squared:  0.8765,	Adjusted R-squared:  0.8724 
F-statistic: 215.2 on 6 and 182 DF,  p-value: < 2.2e-16


In [37]:
jb.norm.test(residuals(mod14))
lillie.test(residuals(mod14))


	Jarque-Bera test for normality

data:  residuals(mod14)
JB = 1.2341, p-value = 0.517



	Lilliefors (Kolmogorov-Smirnov) normality test

data:  residuals(mod14)
D = 0.048823, p-value = 0.3357


--------------------
--------------------
# Estudio de valores influyentes y atipicos

--------
### Influyentes
--------

In [38]:
distancia11 <- cooks.distance(mod11)
distancia12 <- cooks.distance(mod12)
distancia14 <- cooks.distance(mod14)

In [39]:
write.csv(distancia11,file='Distancias de Cook modelo 11.csv')
write.csv(distancia12,file='Distancias de Cook modelo 12.csv')
write.csv(distancia14,file='Distancias de Cook modelo 14.csv')


In [40]:
influyentes11 <- (distancia11 > 2*(7-1)/188)
influyentes12 <- (distancia12 > 2*(10-1)/188)
influyentes14 <- (distancia14 > 2*(6-1)/188)

In [41]:
subset(influyentes11,influyentes11==TRUE)

29  107  162 
TRUE TRUE TRUE

In [42]:
subset(influyentes12,influyentes12==TRUE)

29   69 
TRUE TRUE

In [43]:
subset(influyentes14,influyentes14==TRUE)

29  107  131  162 
TRUE TRUE TRUE TRUE

--------
### Atípicos
--------

In [44]:
#round(head(rstandard(mod11),n=188),d=1)
t<-qt((1-0.025),n-7-1)
atipicos11 = which(abs(rstandard(mod11))>t)
atipicos11

3   4  20  29  87  98 106 107 131 132 135 144 
  3   4  20  29  87  98 106 107 131 132 135 144

In [46]:
#round(head(rstandard(mod12),n=188),d=1)
t<-qt((1-0.025),n-10-1)
atipicos12 = which(abs(rstandard(mod12))>t)
atipicos12

15  42  61  65  69  73  85 117 123 150 180 186 
 15  42  61  65  69  73  85 117 123 150 180 186

In [48]:
#round(head(rstandard(mod14),n=188),d=1)
t<-qt((1-0.025),n-6-1)
atipicos14 = which(abs(rstandard(mod14))>t)

In [49]:
subset(influyentes14,influyentes14==TRUE)
atipicos14

29  107  131  162 
TRUE TRUE TRUE TRUE

20  29  87  98 106 107 131 132 135 144 162 
 20  29  87  98 106 107 131 132 135 144 162

-------------
### Consideraremos los modelos anteriores eliminando de la DB el dato atipico influyente

In [50]:
casen11 <- casen[-c(29),]
casen12 <- casen[-c(69),]
casen14_1 <- casen[-c(29),]

In [51]:
EDIT_modelo11 <- lm(X1 ~ X3 +X4+X6 +V1+V2+R+O-1, data=casen11)
EDIT_modelo12 <- lm(log(X1) ~X2 +X3+X4+X6+V1+V2+Z1+E1+E2+R+O-1, data= casen12)
EDIT_modelo14_1 <- lm(X1 ~ X3+X4+X6+V2+R+O-1, data=casen14_1)

In [52]:
summary(EDIT_modelo11)


Call:
lm(formula = X1 ~ X3 + X4 + X6 + V1 + V2 + R + O - 1, data = casen11)

Residuals:
   Min     1Q Median     3Q    Max 
-91147 -23799   1559  23426 114653 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X3 2.782e-02  5.297e-03   5.251 4.23e-07 ***
X4 2.529e-01  4.171e-02   6.062 7.69e-09 ***
X6 9.451e-02  2.944e-02   3.210  0.00157 ** 
V1 9.129e+03  7.817e+03   1.168  0.24444    
V2 1.490e+04  6.542e+03   2.277  0.02397 *  
R  1.206e+04  5.466e+03   2.206  0.02862 *  
O  2.105e+04  6.665e+03   3.158  0.00186 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 37760 on 180 degrees of freedom
Multiple R-squared:  0.8829,	Adjusted R-squared:  0.8783 
F-statistic: 193.9 on 7 and 180 DF,  p-value: < 2.2e-16


In [53]:
summary(mod11)


Call:
lm(formula = X1 ~ X3 + X4 + X6 + V1 + V2 + R + O - 1, data = casen)

Residuals:
   Min     1Q Median     3Q    Max 
-94294 -25196   1139  22171 112228 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X3 3.065e-02  5.287e-03   5.797 2.95e-08 ***
X4 2.107e-01  3.945e-02   5.340 2.76e-07 ***
X6 9.808e-02  2.993e-02   3.277  0.00126 ** 
V1 1.228e+04  7.869e+03   1.561  0.12028    
V2 1.795e+04  6.561e+03   2.736  0.00685 ** 
R  1.042e+04  5.530e+03   1.885  0.06108 .  
O  2.026e+04  6.777e+03   2.989  0.00319 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 38430 on 181 degrees of freedom
Multiple R-squared:  0.8781,	Adjusted R-squared:  0.8734 
F-statistic: 186.3 on 7 and 181 DF,  p-value: < 2.2e-16


In [54]:
jb.norm.test(residuals(EDIT_modelo11))


	Jarque-Bera test for normality

data:  residuals(EDIT_modelo11)
JB = 2.6803, p-value = 0.211


#### Buscamos nuevos datos atipicos que sean influyentes

In [55]:
distancia11_1 <- cooks.distance(EDIT_modelo11)
influyentes11_1 <- (distancia11_1 > 2*(10-1)/187)
subset(influyentes11_1,influyentes11_1==TRUE)

161 
TRUE

In [81]:
t<-qt((1-0.025),n-7-1)
atipicos11_1 = which(abs(rstandard(EDIT_modelo11))>t)
atipicos11_1

3   4  20  86  97 105 106 130 131 134 143 
  3   4  20  86  97 105 106 130 131 134 143

In [82]:
write.csv(distancia11_1,file='Distancias de Cook modelo 11 MODIFICADO.csv')

### Resultados:
#### 1. No se eliminarán mas datos, pues los datos atípicos restantes no son significativos.
#### 2.  Se mejora el ajuste, con el cambio $R^2 =  0.8781$ --> $0.8829$, y se preserva la normalidad del error.
-------------------------
------------------------

In [57]:
summary(EDIT_modelo12)


Call:
lm(formula = log(X1) ~ X2 + X3 + X4 + X6 + V1 + V2 + Z1 + E1 + 
    E2 + R + O - 1, data = casen12)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.0705 -0.9257  0.1057  1.3042  5.0240 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X2 3.459e-06  8.978e-07   3.853 0.000163 ***
X3 1.368e-06  2.625e-07   5.212 5.19e-07 ***
X4 9.779e-06  2.215e-06   4.414 1.76e-05 ***
X6 2.636e-06  1.480e-06   1.782 0.076540 .  
V1 1.940e+00  4.251e-01   4.564 9.37e-06 ***
V2 2.562e+00  3.321e-01   7.717 8.56e-13 ***
Z1 1.108e+00  3.157e-01   3.510 0.000570 ***
E1 1.993e+00  3.448e-01   5.779 3.34e-08 ***
E2 2.132e+00  4.988e-01   4.273 3.15e-05 ***
R  7.579e-01  2.721e-01   2.785 0.005940 ** 
O  3.110e+00  3.778e-01   8.232 3.99e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.844 on 176 degrees of freedom
Multiple R-squared:  0.9753,	Adjusted R-squared:  0.9738 
F-statistic: 632.2 on 11 and 176 DF,  p-value: < 2.2e-16


In [58]:
summary(mod12)


Call:
lm(formula = log(X1) ~ X2 + X3 + X4 + X6 + V1 + V2 + Z1 + E1 + 
    E2 + R + O - 1, data = casen)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.9742 -0.9237  0.1483  1.3109  5.1271 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X2 3.782e-06  8.922e-07   4.239 3.61e-05 ***
X3 1.473e-06  2.599e-07   5.668 5.78e-08 ***
X4 7.475e-06  1.930e-06   3.872 0.000152 ***
X6 2.745e-06  1.492e-06   1.839 0.067545 .  
V1 1.973e+00  4.287e-01   4.603 7.91e-06 ***
V2 2.550e+00  3.350e-01   7.611 1.55e-12 ***
Z1 1.062e+00  3.177e-01   3.342 0.001015 ** 
E1 1.956e+00  3.475e-01   5.630 6.94e-08 ***
E2 2.118e+00  5.033e-01   4.208 4.08e-05 ***
R  8.210e-01  2.729e-01   3.009 0.003007 ** 
O  3.159e+00  3.805e-01   8.301 2.57e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.861 on 177 degrees of freedom
Multiple R-squared:  0.9749,	Adjusted R-squared:  0.9733 
F-statistic: 624.3 on 11 and 177 DF,  p-value: < 2.2e-16


In [59]:
jb.norm.test(residuals(EDIT_modelo12))


	Jarque-Bera test for normality

data:  residuals(EDIT_modelo12)
JB = 1.3975, p-value = 0.435


#### Buscamos nuevos datos atipicos que sean influyentes

In [72]:
distancia12_1 <- cooks.distance(EDIT_modelo12)
influyentes12_1 <- (distancia12_1 > 2*(10-1)/187)
subset(influyentes12_1,influyentes12_1==TRUE)

29 
TRUE

In [73]:
t<-qt((1-0.025),187-10-1)
atipicos12_1 = which(abs(rstandard(EDIT_modelo12))>t)
atipicos12_1

3   4  15  29  42  61  65  72  84 116 122 149 185 
  3   4  15  29  42  61  65  72  84 116 122 149 185

### Eliminaremos el dato número 29, que es atípico e influyente.

In [71]:
casen12 <- casen[-c(29,69),]
EDIT_modelo12_1 <- lm(log(X1) ~X2 +X3+X4+X6+V1+V2+Z1+E1+E2+R+O-1, data= casen12)

29 
TRUE

In [76]:
summary(EDIT_modelo12_1)


Call:
lm(formula = log(X1) ~ X2 + X3 + X4 + X6 + V1 + V2 + Z1 + E1 + 
    E2 + R + O - 1, data = casen12)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.1130 -0.8967  0.1397  1.2462  4.8646 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X2 3.791e-06  8.971e-07   4.226 3.82e-05 ***
X3 1.211e-06  2.674e-07   4.528 1.10e-05 ***
X4 1.201e-05  2.379e-06   5.048 1.11e-06 ***
X6 2.361e-06  1.465e-06   1.611 0.108936    
V1 1.682e+00  4.334e-01   3.880 0.000148 ***
V2 2.371e+00  3.374e-01   7.027 4.52e-11 ***
Z1 1.214e+00  3.147e-01   3.856 0.000162 ***
E1 1.967e+00  3.405e-01   5.776 3.43e-08 ***
E2 2.287e+00  4.966e-01   4.605 7.91e-06 ***
R  8.032e-01  2.693e-01   2.983 0.003264 ** 
O  3.053e+00  3.737e-01   8.169 5.97e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.82 on 175 degrees of freedom
Multiple R-squared:  0.976,	Adjusted R-squared:  0.9745 
F-statistic: 646.4 on 11 and 175 DF,  p-value: < 2.2e-16


In [79]:
jb.norm.test(residuals(EDIT_modelo12_1))


	Jarque-Bera test for normality

data:  residuals(EDIT_modelo12_1)
JB = 0.78465, p-value = 0.649


In [78]:
distancia12_2 <- cooks.distance(EDIT_modelo12_1)
influyentes12_2 <- (distancia12_2 > 2*(10-1)/186)
subset(influyentes12_2,influyentes12_2==TRUE)

In [83]:
write.csv(distancia12_2,file='Distancias de Cook modelo 12 MODIFICADO.csv')

### Resultados
#### 1. No se eliminarán mas datos, pues los datos atípicos restantes no son significativos
#### 2. Se mejora el ajuste, con el cambio $R^2 =  0.9749$ ---> $0.9753$ ---> $0.976$, y se preserva la normalidad del error.
--------------------
----------------

In [63]:
summary(mod14)


Call:
lm(formula = X1 ~ X3 + X4 + X6 + V2 + R + O - 1, data = casen)

Residuals:
    Min      1Q  Median      3Q     Max 
-102450  -24292    3237   23330  110070 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X3 3.051e-02  5.307e-03   5.749 3.73e-08 ***
X4 2.151e-01  3.950e-02   5.445 1.66e-07 ***
X6 1.067e-01  2.953e-02   3.613 0.000392 ***
V2 1.205e+04  5.384e+03   2.238 0.026454 *  
R  1.149e+04  5.509e+03   2.085 0.038468 *  
O  2.485e+04  6.129e+03   4.055 7.44e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 38580 on 182 degrees of freedom
Multiple R-squared:  0.8765,	Adjusted R-squared:  0.8724 
F-statistic: 215.2 on 6 and 182 DF,  p-value: < 2.2e-16


----------------------------------

In [64]:
summary(EDIT_modelo14_1)


Call:
lm(formula = X1 ~ X3 + X4 + X6 + V2 + R + O - 1, data = casen14_1)

Residuals:
   Min     1Q Median     3Q    Max 
-91850 -25519   1894  24258 113236 

Coefficients:
    Estimate Std. Error t value Pr(>|t|)    
X3 2.754e-02  5.297e-03   5.198 5.39e-07 ***
X4 2.587e-01  4.145e-02   6.242 2.98e-09 ***
X6 1.005e-01  2.901e-02   3.466  0.00066 ***
V2 1.041e+04  5.304e+03   1.964  0.05110 .  
R  1.294e+04  5.420e+03   2.387  0.01801 *  
O  2.444e+04  6.006e+03   4.069 7.04e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 37790 on 181 degrees of freedom
Multiple R-squared:  0.882,	Adjusted R-squared:  0.8781 
F-statistic: 225.5 on 6 and 181 DF,  p-value: < 2.2e-16


In [65]:
jb.norm.test(residuals(EDIT_modelo14_1))


	Jarque-Bera test for normality

data:  residuals(EDIT_modelo14_1)
JB = 2.4647, p-value = 0.2365


#### Buscamos nuevos datos atipicos que sean influyentes

In [66]:
distancia14_1 <- cooks.distance(EDIT_modelo14_1)
influyentes14_1 <- (distancia14_1 > 2*(6-1)/187)
subset(influyentes14_1,influyentes14_1==TRUE)

106  130  161 
TRUE TRUE TRUE

In [85]:
t<-qt((1-0.025),187-6-1)
atipicos14_1 = which(abs(rstandard(EDIT_modelo12))>t)
atipicos14_1

3   4  15  29  42  61  65  72  84 116 122 149 185 
  3   4  15  29  42  61  65  72  84 116 122 149 185

In [84]:
write.csv(distancia14_1,file='Distancias de Cook modelo 14 MODIFICADO.csv')

### Resultado:
#### 1. No se eliminan mas datos, los atipicos restantes no son significativos.
#### 2. Se mejora el ajuste con $R^2 = 0.8765$ ---> $0.882$ y se preserva la normalidad de los residuales.